src/momento/utilities/expiration.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+import math
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import Optional
+
+
+class Expiration(ABC):
+    """Represents an expiration time for a token."""
+
+    _does_expire: bool
+
+    def __init__(self, does_expire: bool) -> None:
+        self._does_expire = does_expire
+
+    @abstractmethod
+    def does_expire(self) -> bool:
+        pass
```

Comment:
The `Expiration` class can be a normal base class, as opposed to an `ABC`. This method doesn't need to be abstract since the derived classes implement in the same way. We can lift that logic up here.

Once you do that you can just make `Expiration` a dataclass.